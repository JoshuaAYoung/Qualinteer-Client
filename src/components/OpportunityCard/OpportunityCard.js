import React, { Component } from 'react';
import { cleanDate } from '../../utilities/Utils';


export default class OpportunityCard extends Component {
  truncate = (str, num_words) => {
    return str.split(' ').splice(0, num_words).join(' ');
  }

  stripString = (descriptionHtml) => {
    let truncatedString = this.truncate(descriptionHtml, 12)
    return truncatedString.replace(/(<([^>]+)>)/ig, "")
  }

  render() {
    const { opportunity_id, title, description, posted, name, city, state, zipcode } = this.props.opportunity
    return (
      <>
        <a href={`/opportunity/${opportunity_id}`} className='oppCardLink'>
          <h3 className='opportunityTitle'>{title}</h3>
          <p className='orgTitle'>{name}</p>
          <p className='orgInfo'>{city}, {state} {zipcode}</p>
          <p className='opportunityDescription'><span className='defaultMedium'>Description: </span> {this.stripString(description)}...  <span className='defaultMedium'>More ></span></p>
          <p className='opportunityPosted'><span className='defaultMedium'>Posted </span>{cleanDate(posted)}</p>
        </a>
      </>
    );
  }
}