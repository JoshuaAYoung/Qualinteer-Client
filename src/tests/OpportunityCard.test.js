import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import OpportunityCard from '../components/OpportunityCard/OpportunityCard'
import VableContext from '../context/VableContext'

describe('OpportunityCard Component', () => {
  const props = {
    opportunity: {
      opportunity_id: 1,
      title: 'test title',
      description: 'test description',
      posted: '2019-12-22T16:28:32.615Z',
      name: 'test name',
      city: 'test city',
      state: 'test state',
      zipcode: 90210
    }
  }

  it('OpportunityCard Smoke Test', () => {
    const div = document.createElement('div')

    ReactDOM.render(<OpportunityCard {...props} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('OpportunityCard Snapshot Test', () => {
    const wrapper = shallow(<OpportunityCard {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})