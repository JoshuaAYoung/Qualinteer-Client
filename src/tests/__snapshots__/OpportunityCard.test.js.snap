// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpportunityCard Component OpportunityCard Snapshot Test 1`] = `
<Fragment>
  <a
    className="oppCardLink"
    href="/opportunity/1"
  >
    <h3
      className="opportunityTitle"
    >
      test title
    </h3>
    <p>
      test name
    </p>
    <p
      className="orgInfo"
    >
      test city
      , 
      test state
       
      90210
    </p>
    <p
      className="opportunityDescription"
    >
      <span
        className="defaultMedium"
      >
        Description: 
      </span>
       
      test description
      ...  
      <span
        className="defaultMedium"
      >
        More &gt;
      </span>
    </p>
    <p
      className="opportunityPosted"
    >
      <span
        className="defaultMedium"
      >
        Posted 
      </span>
      12-22-2019
    </p>
  </a>
</Fragment>
`;
